core/src/main/java/org/apache/iceberg/InheritableMetadataFactory.java

```diff
@@ -46,18 +49,28 @@ static InheritableMetadata forCopy(long snapshotId) {
     private final int specId;
     private final long snapshotId;
     private final long sequenceNumber;
+    private final String tableLocation;
+    private final Map<String, String> tableProperties;
 
-    private BaseInheritableMetadata(int specId, long snapshotId, long sequenceNumber) {
+    private BaseInheritableMetadata(int specId, long snapshotId, long sequenceNumber, String tableLocation,
+        Map<String, String> tableProperties) {
       this.specId = specId;
       this.snapshotId = snapshotId;
       this.sequenceNumber = sequenceNumber;
+      this.tableLocation = tableLocation;
+      this.tableProperties = tableProperties;
     }
 
     @Override
     public <F extends ContentFile<F>> ManifestEntry<F> apply(ManifestEntry<F> manifestEntry) {
       if (manifestEntry.file() instanceof BaseFile) {
         BaseFile<?> file = (BaseFile<?>) manifestEntry.file();
         file.setSpecId(specId);
+        if (MetadataPaths.useRelativePath(tableProperties)) {
+          if (!file.path().toString().startsWith(tableLocation)) {
```

Comment:
Why do not we push this logic inside the `otAbsolutePath` method?

This is only a question which we might to consider, I myself not sure about the answer yet.

Reply:
You are right. Pushed the logic into `toAbsolutePath()`.